src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -251,6 +252,14 @@ public static bool TryComputeNamespace(this RazorCodeDocument document, bool fal
             throw new ArgumentNullException(nameof(document));
         }
 
+        @namespace = (string)document.Items[NamespaceKey];
+
+        // If we cached the result, then we're done!
+        if (@namespace is not null)
```

Comment:
Let's move the rest of this method into a local function (except the cache action) and assert that the computed value matches what's in the cache in DEBUG. This is one of the many examples of `ItemsCollection` usages that feels extremely icky to me, as the thread-affinity here seems completely arbitrary and un-specified.

Reply:
A fair concern, and I will do so, though its worth mentioning the logic to compute the namespace calls about a half dozen methods that all pull state from `ItemsCollection` so I don't consider this introducing a _new_ problem.